web/impact/impact/permissions.py

```diff
@@ -39,17 +49,11 @@ def has_permission(self, request, view):
             name=settings.V1_CONFIDENTIAL_API_GROUP).exists()
 
 
-class DynamicModelPermissions(BasePermission):
-    perms_map = {
-        'GET': ['%(app)s.view_%(model_name)s'],
-        'OPTIONS': ['%(app)s.view_%(model_name)s'],
-        'HEAD': ['%(app)s.view_%(model_name)s'],
-        'POST': ['%(app)s.add_%(model_name)s'],
-        'PUT': ['%(app)s.change_%(model_name)s'],
-        'PATCH': ['%(app)s.change_%(model_name)s'],
-        'DELETE': ['%(app)s.delete_%(model_name)s'],
-    }
+def method_to_perm(method):
+    return ["%(app)s.{}_%(model_name)s".format(METHOD_TO_ACTION[method])]
```

Comment:
It's a little mind-bending to have both old- and new-style string interpolation used in a single line. I'd prefer it to be all new-style.

Reply:
I had trouble the last time I tried to do one replacement without affecting the other things that will be replaced later when I used format.  I expect there's some trick for it, but I gave up searching before I found it.  Do you know how to do that?  The other option would be to just use `+`, but I'm not sure that's any better that what I'm doing now.

Reply:
Found this on SO, but I'm not lovin' it: https://stackoverflow.com/questions/11283961/partial-string-formatting

Reply:
Aha yeah, I think `string.Template` is the only built-in way to do this. Not a win.
```python
>>> from string import Template
>>> t1 = Template("${app}.${action}_${model_name}")
>>> t2 = t1.safe_substitute(action="edit")
>>> print Template(t2).safe_substitute(app="admin", model_name="logentry")
admin.edit_logentry
```
Aesthetically I'd prefer concatenation to the interpolation mashup, but that preference isn't strong enough to delay approving this.